stan/math/prim/fun/acosh.hpp

```diff
@@ -83,8 +83,9 @@ struct acosh_fun {
  * @param x container
  * @return Elementwise acosh of members of container.
  */
-template <typename T, require_all_not_nonscalar_prim_or_rev_kernel_expression_t<
-                          T>* = nullptr>
+template <
```

Comment:
Actually this can just be `template <typename T, require_container_t<T>* = nullptr>`

Reply:
There were some functions here that didn't have `apply_vector_unary`/Container implementations. I'm assuming it's because the Eigen array functions didn't exist for them to be written (`cbrt`, this, and others). I don't think cleanup is what we're going for in this pull really, so it's fine to leave it.

Reply:
(But if you want to clean things, feel free, happy to review)

Reply:
Ups actually ignore that, this needs to be able to take in scalar types so that's why we can't use `require_constainer_st`